search/includes/classes/class-versioningcleanupjob.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Automattic\VIP\Search;
+
+class VersioningCleanupJob {
+
+	/**
+	 * The name of the scheduled cron event to run the versioning cleanup.
+	 */
+	const CRON_EVENT_NAME = 'vip_search_versioning_cleanup';
+
+	const WEEK_IN_SECONDS = 7 * 24 * 60 * 60;
+
+	const SEARCH_ALERT_SLACK_CHAT = '#vip-go-es-alerts';
+
+	public function __construct( $indexables, $versioning ) {
```

Comment:
Hmm I am not sure how to pass the dependencies correctly. But doing this felt the most correct. However, I haven't seen this pattern anywhere in our code so there is possibly a better way?